src/include/simeng/arch/aarch64/helpers/auxiliaryFunctions.hh

```diff
@@ -265,16 +265,22 @@ class AuxFunc {
   static uint16_t sveGetPattern(const std::string operandStr,
                                 const uint8_t esize, const uint16_t VL_) {
     const uint16_t elements = VL_ / esize;
-    // If not pattern then same as ALL
-    if (operandStr.find(",") == std::string::npos) return elements;
+    const std::vector<std::string> patterns = {
+        "pow2", "vl1",  "vl2",  "vl3",   "vl4",   "vl5",  "vl6",  "vl7", "vl8",
+        "vl16", "vl32", "vl64", "vl128", "vl256", "mul3", "mul4", "all"};
 
-    // Get pattern string
-    std::string pattern(operandStr.substr(operandStr.find(",") + 2));
-    if (pattern.find(',') != std::string::npos) {
-      pattern.erase(pattern.find(","));
+    // If not pattern then same as ALL
```

Comment:
Comment should be changed to be more easily read - something like "If no pattern present in operandStr then same behaviour as ALL"